docs/src/main/asciidoc/grpc-service-consumption.adoc

```diff
@@ -401,6 +405,47 @@ To disable the gRPC client metrics when `quarkus-micrometer` is used, add the fo
 quarkus.micrometer.binder.grpc-client.enabled=false
 ----
 
+== Custom exception handling
+
+If any of the gRPC services or server interceptors throw an (custom) exception, you can add your own https://github.com/quarkusio/quarkus/extensions/grpc/api/src/main/java/io/quarkus/grpc/ExceptionHandlerProvider.java[ExceptionHandlerProvider]
+as a CDI bean in your application, to provide a custom handling of those exceptions.
+
+e.g.
+
+[source, java]
+----
+@ApplicationScoped
+public class HelloExceptionHandlerProvider implements ExceptionHandlerProvider {
+    @Override
+    public <ReqT, RespT> ExceptionHandler<ReqT, RespT> createHandler(ServerCall.Listener<ReqT> listener,
+            ServerCall<ReqT, RespT> serverCall, Metadata metadata) {
+        return new HelloExceptionHandler<>(listener, serverCall, metadata);
+    }
+
+    @Override
+    public Throwable transform(Throwable t) {
```

Comment:
Should the returned type be a StatusRuntime Exception?

Reply:
No need + there is a problem of which one to return -- StatusRuntimeException or plain StatusException ...
I guess users will mostly return one of them ... but they can opt into other runtime exceptoins as well, if they want them to propagate up the stack ...